### PR TITLE
Indication of Search Results + No Next Page Button On Last Page

### PR DESCRIPTION
**Pull request by topperblue**
Text "No results found." displays upon search returning no results.

The "Next Page" button does not appear on the last page of posts.

### CODE REVIEW

File: client/src/components/Posts/index.js [hunk 1]
```diff
@@ -9,19 +9,25 @@ import NextPageButton from '../NextPageButton';
...
-        {posts.map(post => (
-          <Post data={post} key={post.url} />
-        ))}
-
-        {(!lastPage) ? <NextPageButton /> : ''}
-    </div>
-  );
+  if (posts.length > 0) {
+    return (
+      <div className='Posts' id='Posts'>
```
Comment (SkyIsTheLimit) [inline client/src/components/Posts/index.js:14]: Is the id attribute used somewhere else?

File: client/src/components/Posts/index.js [hunk 2]
```diff
@@ -9,19 +9,25 @@ import NextPageButton from '../NextPageButton';
...
+          {posts.map(post => (
+            <Post data={post} key={post.url} />
+          ))}
+          
+          {(!lastPage) ? <NextPageButton /> : ''}
+      </div>
+    );
+  } else {
+    return (
+      <div className='NoResults' id='NoResults'>No results found.</div>
```
Comment (SkyIsTheLimit) [inline client/src/components/Posts/index.js:27]: Is the id attribute used somewhere else?

File: server/src/api/post/index.js [hunk 3]
```diff
@@ -124,15 +124,16 @@ routes.post('/:page', async (req, res) => {
...
+  const filteredPostCount = filteredPosts.length;
   const skipCount = pageNum * pageSize;
   const posts = await postsCollection
     .find(filter)
     .sort(sortParam)
     .skip(pageNum * pageSize)
     .limit(pageSize)
     .toArray();
-  const lastPage = posts.length === 0 || (postCount - (skipCount + posts.length)) <= 0;
+  const lastPage = posts.length === 0 || (filteredPostCount - (skipCount + posts.length)) <= 0;
```
Comment (SkyIsTheLimit) [inline server/src/api/post/index.js:136]: This is brilliant 🤣

File: client/src/components/Posts/index.js (hunk 1 above)
Comment (topperblue) [reply client/src/components/Posts/index.js:14]: No, I just kept it there for posterity

File: client/src/components/Posts/index.js (hunk 2 above)
Comment (topperblue) [reply client/src/components/Posts/index.js:27]: No

**Review by SkyIsTheLimit — APPROVED** [review]

**Review by codelastnight — APPROVED** [review]
fixed merge conflicts 👍

**Review by codelastnight — APPROVED** [review]

**Pull request merged**